### DISCUSSION TITLE
Why can't RYD just show the raw data?

### DISCUSSION THREAD

**Discussion by agausmann** [Anarios/return-youtube-dislike#1002] [General]
As I understand it now, RYD uses its database of votes (the "raw" data) as well as public data to extrapolate/predict the real dislike count, and that's all that it displays in the webpage.

Why is RYD so concerned about reconciling one data source with another data source? Why can't it just show its own like/dislike counts separate from YouTube's system?

**Comment by jpa102** [comment]
around august 4, 2024, [there had been some discussion about dislike data regarding mrbeast's video](https://github.com/Anarios/return-youtube-dislike/issues/1066)

from this time period, 2 new fields in the json response has been introduced:

* `rawLikes`
* `rawDislikes`

it will look like this: (example url is https://returnyoutubedislikeapi.com/votes?videoid=YbJOTdZBX1g)

```
{
	"id":"YbJOTdZBX1g",
	"dateCreated":"2022-04-09T22:01:29.366144Z",
	"likes":3164935,
	"rawDislikes":136928,
	"rawLikes":8555,
	"dislikes":20457546,
	"rating":1.5359191526072133,
	"viewCount":233544526,
	"deleted":false
}
```

i have adapted my site to use these 2 new fields, to display raw likes & dislikes separate from actual likes & estimated dislikes

**Comment by agausmann** [comment]
Thanks for the update, it looks like things are going in the right direction.
I read that discussion and have some more points I'd like to bring up in response, to expand upon my initial post:

[unknown source]:
> RYD Dislikes = (RYD users Likes / RYD users dislikes) * actual video Likes
>
> also an approximation is better than nothing, but you should not blame the extension, it was never meant to be 100% accurate.

ParashRahman replied:
> This app needs to be downloaded and installed - it's a different audience who would do that. Many people dont care about this app. However, for that audience, it's a good estimate.

Anarios further clarified:
> The formula is also an oversimplification - there are weights applied, aimed at preventing bias (i.e. an attempt to counter the difference between the extension users and the users in general).

The problem with all of this is, we don't know the size of the non-RYD audience that shares sentiments with the RYD userbase. Whatever the formula is, it is trying to speak for viewers outside of RYD, and is always going to be inaccurate to some extent and misrepresent them.

In my opinion, RYD should just treat itself as a self-contained community and show only the likes/dislikes inside the community, _not_ try to extrapolate its information to other people that it knows very little about and inevitably overrepresent the sentiments of RYD users. (Sidenote: this sounds a lot like an echo chamber effect)

A good example of this is Grayjay - their like/dislike buttons show only the counts that are in Grayjay's database, and the YouTube like count (and optionally RYD's dislike estimate) are displayed next to the buttons. Currently, very few videos get likes/dislikes on Grayjay, but whatever the data is, I am assured that they are real people, and the magnitude difference between Grayjay and YouTube is a constant reminder that the data is limited to that community.

I appreciate the effort that @Anarios puts in to trying to make the estimates more accurate. It is clear that he takes responsibility and tries to minimize the misrepresentation. My opinion is that those efforts are misplaced, it would be better to just be transparent about RYD's sample size by showing the like/dislike counts of just the RYD community.

**Reply by jpa102** [reply]
> we don't know the size of the non-RYD audience that shares sentiments with the RYD userbase

fair enough. combining the estimated total users of this extension, there are 5.7M users (+ few million more if we count those who have unsupported / unofficial way of using its api to view dislikes)

> RYD should just treat itself as a self-contained community and show only the likes/dislikes inside the community

this is something that (i think?) i've seen in closed github issues, they were requesting to show the dislike counts made by extension users

with the `rawLikes` and `rawDislikes` fields in the json response being present now, this could be implemented easily

> I appreciate the effort that @Anarios puts in to trying to make the estimates more accurate.

yeah, i'm glad that person is doing everything to make the estimated dislike counts as "fair" as possible. i've seen some videos that have unfair amount of estimated dislikes, even though the comment section tells you otherwise

**Comment by Anarios** [comment]
You can now see RYD’s raw like and dislike counts separately in the side panel, available since **4.0.3**. The latest release is [v4.0.5](https://github.com/Anarios/return-youtube-dislike/releases/tag/v4.0.5).

<img width="1897" height="867" alt="Raw like and dislike counts in the side panel" src="https://github.com/user-attachments/assets/a4206563-83d5-4e2a-aebd-1ebffa2f76af" />
